### PR TITLE
OSD-14612 - Strip down the SREP support role policy some more

### PR DESCRIPTION
**Pull request by wshearn**
### What type of PR is this?
cleanup

### What this PR does / why we need it?
This PR is stripping more permissions from the SREP policy.

### Which Jira/Github issue(s) this PR fixes?
[OSD-14612](https://issues.redhat.com//browse/OSD-14612)/OSD-15458

_Fixes #_

### Pre-checks (if applicable):
- [ ] Tested latest changes against a cluster
- [ ] Included documentation changes with PR
- [ ] If this is a new object that is not intended for the FedRAMP environment (if unsure, please reach out to team FedRAMP), please exclude it with:

    ```yaml
    matchExpressions:
    - key: api.openshift.com/fedramp
      operator: NotIn
      values: ["true"]
    ```

### CODE REVIEW

File: resources/sts/4.12/hypershift/sts_hcp_support_permission_policy.json [hunk 1]
```diff
@@ -32,8 +26,6 @@
         "s3:GetObjectTagging",
         "s3:ListAllMyBuckets",
         "sts:DecodeAuthorizationMessage",
```
Comment (karthikperu7) [inline resources/sts/4.12/hypershift/sts_hcp_support_permission_policy.json:28]:
```suggestion
        "sts:DecodeAuthorizationMessage"
```

File: resources/sts/4.12/hypershift/sts_hcp_support_permission_policy.json [hunk 2]
```diff
@@ -65,36 +57,6 @@
         "arn:aws:s3:::*image-registry*"
```
Comment (karthikperu7) [inline resources/sts/4.12/hypershift/sts_hcp_support_permission_policy.json:57]: I think managed-velero* should be removed here as velero is not a thing in HyperShift

File: resources/sts/4.12/hypershift/sts_hcp_support_permission_policy.json [hunk 3]
```diff
@@ -65,36 +57,6 @@
...
-      "Resource": "arn:aws:ec2:*:*:network-insights-analysis/*",
-      "Condition": {
-        "StringEquals": {
-          "aws:ResourceTag/red-hat-managed": "true"
-        }
-      }
-    },
     {
       "Effect": "Allow",
       "Action": "ec2:RunInstances",
```
Comment (karthikperu7) [inline resources/sts/4.12/hypershift/sts_hcp_support_permission_policy.json:62]: this is minor, but the indentation in this file looks a little off. would be nice to get them all consistent

Comment (wshearn) [reply resources/sts/4.12/hypershift/sts_hcp_support_permission_policy.json:62]: I have formatted it so the tab spaces matches the other files. Should we define a formatting guide for json files so we have an easy way to keep them uniform?

File: resources/sts/4.12/hypershift/sts_hcp_support_permission_policy.json [hunk 4]
```diff
@@ -1,341 +1,292 @@
...
-        "kms:CreateGrant",
-        "route53:GetHosted*",
-        "route53:List*",
-        "s3:GetBucketTagging",
-        "s3:GetObjectAcl",
-        "s3:GetObjectTagging",
-        "s3:ListAllMyBuckets",
-        "sts:DecodeAuthorizationMessage",
-        "tiros:CreateQuery",
-        "tiros:GetQuery*"
```
[the remark this answers is not in the document]
Comment (wshearn) [reply resources/sts/4.12/hypershift/sts_hcp_support_permission_policy.json:36]: Added back, though it makes me wonder if there is a better way to handle this then. I am guessing no unless some code changes to the network analyzer.

File: resources/sts/4.12/hypershift/sts_hcp_support_permission_policy.json [hunk 5]
```diff
@@ -1,341 +1,294 @@
...
+                "ec2:Describe*",
+                "ec2:GetTransitGatewayAttachmentPropagations",
+                "ec2:GetTransitGatewayMulticastDomainAssociations",
+                "ec2:GetTransitGatewayPrefixListReferences",
+                "ec2:GetTransitGatewayRouteTableAssociations",
+                "ec2:GetTransitGatewayRouteTablePropagations",
+                "elasticloadbalancing:Describe*",
+                "iam:GetRole",
+                "iam:ListRoles",
+                "kms:CreateGrant",
```
Comment (jharrington22) [inline resources/sts/4.12/hypershift/sts_hcp_support_permission_policy.json:21]: This should be scoped now once PR #1546 merges

**Review by jharrington22 — COMMENTED** [review]
What is the character size for this policy? I'm wondering if we can expand the `ec2:Describe*` or not

**Comment by dustman9000** [comment]
Verified https://github.com/openshift/managed-cluster-config/pull/1546 incorporated properly.
/lgtm
/approve

File: resources/sts/4.12/hypershift/sts_hcp_support_permission_policy.json (hunk 5 above)
Comment (wshearn) [reply resources/sts/4.12/hypershift/sts_hcp_support_permission_policy.json]: I have pulled these changes in due to merge conflicts.

**Comment by jharrington22** [comment]
@wshearn @karthikperu7 can you confirmed what was actually removed from this policy?

**Pull request merged**